cmd/manager/main.go

```diff
@@ -142,7 +155,32 @@ func main() {
 		os.Exit(1)
 	}
 
-	if err = serveCRMetrics(cfg); err != nil {
+	// Add the Metrics Service
+	addMetrics(ctx, cfg)
+
+	log.Info("Starting the Cmd.")
+
+	// Start the Cmd
+	if err := mgr.Start(signals.SetupSignalHandler()); err != nil {
+		log.Error(err, "Manager exited non-zero")
+		os.Exit(1)
+	}
+}
+
+
+// addMetrics will create the Services and Service Monitors to allow the operator export the metrics by using
+// the Prometheus operator
+func addMetrics(ctx context.Context, cfg *rest.Config) {
+	// Get the namespace the operator is currently deployed in.
+	operatorNs, err := k8sutil.GetOperatorNamespace()
+	if err != nil {
+		if errors.Is(err, k8sutil.ErrRunLocal) {
```

Comment:
This also needs to check for `ErrNoNamespace`.

Reply:
This was copied directly from the operator-sdk upgrade docs.  My understanding is that the service isn't created if running locally, with `OSDK_FORCE_RUN_MODE=local`.  If `ErrNoNamespace` is returned, it means the operator is not running locally and is not running in a namespace (?).

I've updated `common.GetOperatorNamesapce()` to handle the `ErrRunLocal` error instead of `ErrNoNamespace` error, so local debugging should work if you set the env `OSDK_FORCE_RUN_MODE=local`

Reply:
Looks like `ErrNoNamespace` was required for the unit tests, so I've added that back (or'd with `ErrRunLocal`).